canal/canal.go

```diff
@@ -20,13 +24,54 @@ import (
 	"github.com/siddontang/go-mysql/schema"
 )
 
+func init() {
+	ast.NewValueExpr = newValueExpr
+	ast.NewParamMarkerExpr = newParamExpr
+	ast.NewDecimal = func(_ string) (interface{}, error) {
+		return nil, nil
+	}
+	ast.NewHexLiteral = func(_ string) (interface{}, error) {
+		return nil, nil
+	}
+	ast.NewBitLiteral = func(_ string) (interface{}, error) {
+		return nil, nil
+	}
+}
+
+type paramExpr struct {
+	valueExpr
+}
+
+func newParamExpr(_ int) ast.ParamMarkerExpr {
+	return &paramExpr{}
+}
+func (pe *paramExpr) SetOrder(o int) {}
+
+type valueExpr struct {
+	ast.TexprNode
+}
+
+func newValueExpr(_ interface{}) ast.ValueExpr                      { return &valueExpr{} }
+func (ve *valueExpr) SetValue(val interface{})                      {}
+func (ve *valueExpr) GetValue() interface{}                         { return nil }
+func (ve *valueExpr) GetDatumString() string                        { return "" }
+func (ve *valueExpr) GetString() string                             { return "" }
+func (ve *valueExpr) GetProjectionOffset() int                      { return 0 }
+func (ve *valueExpr) SetProjectionOffset(offset int)                {}
+func (ve *valueExpr) Restore(ctx *format.RestoreCtx) error          { return nil }
+func (ve *valueExpr) Accept(v ast.Visitor) (node ast.Node, ok bool) { return }
+func (ve *valueExpr) Text() string                                  { return "" }
+func (ve *valueExpr) SetText(text string)                           {}
+func (ve *valueExpr) Format(w io.Writer)                            {}
+
 // Canal can sync your MySQL data into everywhere, like Elasticsearch, Redis, etc...
 // MySQL must open row format for binlog
 type Canal struct {
 	m sync.Mutex
 
 	cfg *Config
 
+	parser     *parser.Parser
```

Comment:
it's better to use another variable name rather than `parser` to avoid conflict with `parser` pkg name.

Reply:
it fine to use this name, because it is a member of struct, and would not be use directly. so not conflict can hanppend.